src/algebra/group/pi.lean

```diff
@@ -32,6 +32,19 @@ by refine_struct { mul := (*), .. }; tactic.pi_instance_derive_field
 instance monoid [∀ i, monoid $ f i] : monoid (Π i : I, f i) :=
 by refine_struct { one := (1 : Π i, f i), mul := (*), .. }; tactic.pi_instance_derive_field
 
+@[simp]
+lemma single_zero [∀ i, add_monoid $ f i] [decidable_eq I] (i : I) :
+  single i (0 : f i) = 0 :=
+function.update_eq_self i 0
+
+lemma single_add [∀ i, add_monoid $ f i] [decidable_eq I] (i : I) (x y : f i) :
```

Comment:
Should these lemmas be simp-lemmas? The corresponding lemmas for bundled homs are simp.

Reply:
My memory is that I usually want to collect as many terms as possible under a single call to `single` - so if these were simp lemmas, I'd want them in the reverse direction.

My memory may be wrong here.

Reply:
Ok, I'm fine with being conservative here (-;